apps/notifications/src/doc/note-rendering.md

```diff
@@ -0,0 +1,105 @@
+# Notification Detail Rendering
+
+When a user clicks a note to view, that notification is rendered to the user in a slide-out drawer. The pipeline for rendering this drawer to the user begins in `/src/panel/templates/index.jsx`. When the user has selected a note to view, this component renders a child `Note` component from `src/panel/templates/note.jsx` with the property `detailView` set to `true`.
+
+This `detailView` prop further causes the `Note` component to render (among other things) a `NoteBody` component (`/src/panel/templates/body.jsx`). The `render` method for this component is where the majority of rendering logic for a notification takes place.
+
+## Rendering Pipeline
+
+The notification object returned from the server contains, among other things, a `body` array of one or more `block` objects which contain a type, the text of the notification, and an array of `ranges`:
+
+```js
+"ranges": [
+	{
+		"url": "https://www.example.com/",
+		"indices": [
+			397,
+			416
+		]
+	},
+	{
+		"type": "match",
+		"indices": [
+			598,
+			616
+		]
+	},
+	{
+		"type": "match",
+		"indices": [
+			682,
+			700
+		]
+	}
+]
+```
+
+The indices of these ranges indicate locations in the body's text string where special formatting is needed.
+
+After some initial processing, the `NoteBody` component performs a switch statement on each block object based on its type:
+
+```js
+switch ( block.signature.type ) {
```

Comment:
Is adding the code here necessary? It may be sufficient to just refer to the source code. Adding the code here results in two places where this needs to be maintained

Reply:
Good call -- I'll tweak that. :) Thanks.